crates/misc/dotnet/src/WasmtimeException.cs

```diff
@@ -20,5 +21,27 @@ public class WasmtimeException : Exception
 
         /// <inheritdoc/>
         protected WasmtimeException(SerializationInfo info, StreamingContext context) : base(info, context) { }
+
+        internal static WasmtimeException FromOwnedError(IntPtr error)
+        {
+            unsafe
+            {
+                Interop.wasmtime_error_message(error, out var bytes);
+                var byteSpan = new ReadOnlySpan<byte>(bytes.data, checked((int)bytes.size));
+
+                int indexOfNull = byteSpan.LastIndexOf((byte)0);
```

Comment:
Is this necessary?  From the C API trap messages are null terminated (for some reason...), but for our API we don't have to do that I assume.